src/Util/Search.php

```diff
@@ -765,4 +790,29 @@ private function doesUdiExistInQueryTerm(string $queryTerm, Query\BoolQuery $fie
         }
         return $queryTerm;
     }
+
+    /**
+     * Split the query terms into must match and must not match terms.
+     *
+     * @param string $queryTerm Query term that needs to be searched upon.
+     *
+     * @return array
+     */
+    private function splitQueryTerms(string $queryTerm): array
+    {
+        $splitUpQueryTerms = array();
+        if (preg_match_all('/\s-\b(\w*)\b/', $queryTerm, $matches)) {
+            $splitUpQueryTerms = array(
+                'mustNotMatch' => '',
+                'mustMatch' => ''
+            );
+            foreach ($matches[1] as $match) {
+                $splitUpQueryTerms['mustNotMatch'] = trim($splitUpQueryTerms['mustNotMatch'] . ' ' . $match);
```

Comment:
Did you mean to append here instead?

Reply:
I'm not sure what you are asking

Reply:
Wouldn't this result in only the very last term in the match array being used?

Reply:
I see it now...my bad...